tests/Helper/ObjectParserTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\Tests\Helper;
+
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Validator\Helper\ObjectParser;
+
+final class ObjectParserTest extends TestCase
+{
+    public function dataSkipStaticProperties(): array
+    {
+        return [
+            [
+                ['a' => 4, 'b' => 2],
+                new class () {
+                    public int $a = 4;
+                    public static int $b = 2;
```

Comment:
Let's add non-static "$c" property here to make sure the loop doesn't stop and continues.